src/widgets/IconButton.js

```diff
@@ -0,0 +1,96 @@
+/* eslint-disable react/forbid-prop-types */
+/**
+ * mSupply Mobile
+ * Sustainable Solutions (NZ) Ltd. 2019
+ */
+
+import React from 'react';
+import { TouchableOpacity as TouchableHighlight, StyleSheet, View, Text } from 'react-native';
+import PropTypes from 'prop-types';
+
+import globalStyles, { SUSSOL_ORANGE, LIGHT_GREY, GREY } from '../globalStyles/index';
+
+const ICON_SIZES = {
+  small: 10,
+  medium: 15,
+  large: 20,
+  xlarge: 30,
+};
+
+/**
+ * Simple component rendering a button with a circular border radius
+ * with an icon centered.
+ *
+ * @prop {Node} IconComponent An icon component to render in the center of the circle.
+ * @prop {Func} onPress OnPress callback.
+ * @prop {Func} onPressIn onPressIn callback.
+ * @prop {Func} onPressOut  onPressOut callback.
+ * @prop {Boolean} isDisabled  When true, no action on press. Disabled styling too.
+ * @prop {String} size  Use size preset. "small", "medium" or "large".
+ * @prop {String} leftText  Button label text to left of icon.
+ * @prop {String} rightText Button label text to right of icon
+ */
+export const IconButton = ({
+  IconComponent,
```

Comment:
This is cool, just a consideration: I kind of like passing `<SomeIcon/>` as JSX so that you can easily and declaritively pass props as 'intended'

`<IconButton Icon={<CogIcon color="red" size="large"/>} onPress={}/>`

over

```
const SomeNameSimilarToCogIconButIsnt = () => <CogIcon color="red"/>
<IconButton IconComponent={SomeNameSimilarToCogIconButIsnt}/>
```

or

```
<IconButton IconComponent={CogIcon} iconColor="red"/>
```

where you're likely to add `iconStyle` or `iconProps` props


But after typing this out, `Icon` has like 3 props, so whatevs

Reply:
I largely based this off CircleButton (IconButton comment needs updating to hide my copy paste 😛 ), I don't think it paid off (to be fair I was trying to knock that out too, but the props were getting nuts...). At the moment I feel it's not as much code reduction as it is obfuscation. I think a solution like this would arguably be better (this is kind of why I'm unhappy with the direction this PR ended up):

```jsx
<Button onPress={doThing}>
  <CogIcon size={ICON_SIZE.large} color={GREY} /> // Or omit props if the defaults we write in `icons.js` cover 90%-100% of cases
  <AppText>{localisation.do_thing}</AppText>
</Button>
```

That seems really clean, and anyone at any react skill level can walk into mobile, see that and know what's up. I.e. simple AF = maintainable and minimal learning curve.

It'd be great if we did a better react pattern for handling a consistent theme. I think it'd be neat to `useContext` in CogIcon and AppText to get size constants, fonts and what not, determined by the last parent to set them (i.e. Button could have a prop `size`, which it's children pick up). There may be other patterns these days, I'm pretty rusty heh.

Reply:
Just looking at this again, I like:

```jsx
<IconButton Icon={<CogIcon color="red" size="large"/>} onPress={}/>
```

But the example of size prop there would mean making every Icon in `icons.js` have code to handle the size constant. You could do it with a HOC in that manner, but I think it'd maybe be better left to a UI library/theming overhaul?

Reply:
yeah - it might be OK to just do something for the one icon - but this is more of an icons issue and not an issue with iconbutton - just pass the props however the icon expects?

Reply:
Yea, can do but then you're importing (or using context) in every parent component to get the size constants :(

Reply:
I don't think i'm on the same page.

I think just pass `size={x}` and forget the constants - aren't the constants really fixing problems with `icons` rather than with this button?

(Although you might also claim that the sizing of text and icon SHOULD be coupled :))

My head hurts